packages/component-library/src/InitialFocus.ts

```diff
@@ -1,19 +1,36 @@
 import {
-  type ReactElement,
-  type Ref,
+  Children,
   cloneElement,
+  isValidElement,
+  type ReactElement,
+  Ref,
   useEffect,
   useRef,
 } from 'react';
 
+type FocusableElement = HTMLElement;
+
 type InitialFocusProps = {
-  children:
-    | ReactElement<{ inputRef: Ref<HTMLInputElement> }>
-    | ((node: Ref<HTMLInputElement>) => ReactElement);
+  /**
+   * The child element to focus when the component mounts. This can be either a single React element or a function that returns a React element.
+   */
+  children: ReactElement | ((node: Ref<HTMLInputElement>) => ReactElement);
```

Comment:
To fix ref error below and update to allow any element not only inputs:
```suggestion
  children: ReactElement<{ ref: Ref<HTMLElement> }> | ((node: Ref<HTMLElement>) => ReactElement);
```

Reply:
This is still not resolved

Reply:
dont know how I missed that.